x-pack/plugins/apm/typings/elasticsearch/aggregations.ts

```diff
@@ -301,6 +310,23 @@ interface AggregationResponsePart<
       ? Record<string, DateRangeBucket>
       : { buckets: DateRangeBucket[] };
   };
+  auto_date_histogram: {
+    buckets: Array<
+      {
+        doc_count: number;
+        key: number;
+        key_as_string: string;
+      } & BucketSubAggregationResponse<
+        TAggregationOptionsMap['aggs'],
+        TDocument
+      >
+    >;
+    interval: string;
+  };
+
+  percentile_ranks: {
+    values: Record<string, number> | Array<{ key: number; value: number }>;
```

Comment:
This could be based on the `keyed` prop I think? See `date_range` for an example of how to use `TAggregationOptionsMap` to determine the response type.

Reply:
yes it's based on keyed props, sure let me take a look at TAggregationOptionsMap.